Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-auth.tsx

```diff
@@ -14,48 +14,124 @@ import { useUrls } from './use-urls';
     };
 }
 
+const defaultState = {
+    user: {
+        id: '',
+        username: '',
+        email: '',
+        permissions: { canAccessAdministration: false } as IUserPermissionsType,
+    },
+};
+
 const AuthContext = createContext<IAuthContext>({} as IAuthContext);
 
-interface IAuthProviderProps extends IHaveChildrenProps {
-    user: IUserType;
-}
+type IAuthProviderProps = IHaveChildrenProps
 
 interface ILoginDetailsType {
     Username: string;
     Password?: string;
     RememberMe: boolean;
 }
 
-const AuthProvider = ({ user, children }: IAuthProviderProps) => {
+const AuthProvider = ({ children }: IAuthProviderProps) => {
     const { startLoading, stopLoading } = useLoading();
-    const [ internalUser, setInternalUser ] = useState(user);
-    const [ username, setUsername ] = useState<string>(user.username);
+    const [ internalUser, setInternalUser ] = useState<IUserType>(defaultState.user);
+    const [ username, setUsername ] = useState<string>('');
     const [ password, setPassword ] = useState<string>();
     const [ loginErrorMessage, setLoginErrorMessage ] = useState<string>('');
     const { showError } = useNotifications();
     const defaultLoginErrorMessage = useMemo(() => 'Invalid username or password', []);
 
-    const { getLogoutUrl, getLoginSubmitUrl } = useUrls();
+    const { getLogoutUrl, getLoginSubmitUrl, getUserAuthInfoUrl } = useUrls();
 
     const {
         post: loginSubmit,
         response: loginSubmitResponse,
         status: loginSubmitStatus,
     } = useHttp<null, string, ILoginDetailsType>({ url: getLoginSubmitUrl });
 
+    const {
+        get: getAuthInfo,
+        data: authInfo,
+        isSuccess: isGetAuthInfoSuccess,
+        status: getAuthInfoStatus,
+    } = useHttp<null, IUserResponseType, null>({ url: getUserAuthInfoUrl });
+
     const { post: logout } = useHttp({ url: getLogoutUrl });
 
+    const loginOrGetAuthInitiated = useMemo(
+        () => getAuthInfoStatus !== HttpStatus.NotStarted ||
+        loginSubmitStatus !== HttpStatus.NotStarted,
+        [ getAuthInfoStatus, loginSubmitStatus ],
+    );
+
+    const hasCompletedGetAuthInfo = useMemo(
+        () => getAuthInfoStatus !== HttpStatus.NotStarted &&
+            getAuthInfoStatus !== HttpStatus.Pending,
+        [ getAuthInfoStatus ],
+    );
+
+    const isGetAuthInfoUnauthorized = useMemo(
+        () => getAuthInfoStatus === HttpStatus.Unauthorized,
+        [ getAuthInfoStatus ],
+    );
+
+    const getUser = useCallback(
+        () => internalUser,
+        [ internalUser ],
+    );
+
+    const isLoggedIn = useMemo(
+        () => isGetAuthInfoSuccess && !isEmpty(internalUser.id),
+        [ internalUser, isGetAuthInfoSuccess ],
+    );
+
+    const getAuth = useCallback(async () => {
+        if (loginOrGetAuthInitiated || isLoggedIn) {
+            return;
+        }
+
+        await getAuthInfo();
+    }, [ getAuthInfo, isLoggedIn, loginOrGetAuthInitiated ]);
```

Comment:
I see that the same `if` checks are in the `useEffect` where this is called. Is there a need for a separate function `getAuth` in this case, or we can use directly `getAuthInfo()` in the `useEffect`

Reply:
You are right, i'll edit this